src/Perl6/Compiler.nqp

```diff
@@ -93,4 +97,69 @@ class Perl6::Compiler is HLL::Compiler {
         For more information, see the perl6(1) man page.\n"); 
         nqp::exit(0);
     }
+
+    method interactive(*@args, *%adverbs) {
```

Comment:
I'm considering breaking this and perhaps `readline` out into a role/subclass just for interactive `Compiler`s.